station/app/crud/crud_local_train.py

```diff
@@ -303,5 +337,45 @@ def get_last_run(self, db: Session, train_id: str):
         runs = db.query(LocalTrainExecution).filter(LocalTrainExecution.train_id == train_id).all()
         print(runs)
 
+    def _create_config(self, obj_in):
+        if not isinstance(obj_in, (lt_schemas.LocalTrainConfigCreate, lt_schemas.LocalTrainConfigUpdate)):
+            raise HTTPException(status_code=400,
+                                detail=f"obj_in is not of type LocalTrainConfigSchema but of type {type(obj_in)}")
+
+        airflow_config = self._create_empty_config(None)
+        airflow_config["repository"] = self._get_repository(obj_in.image)
+        if obj_in.tag is not None:
+            airflow_config["tag"] = obj_in.tag
+        else:
+            airflow_config["tag"] = "latest"
+
+        airflow_config["query"] = obj_in.query
+        airflow_config["entrypoint"] = obj_in.entrypoint
+        airflow_config["volumes"] = obj_in.volumes
+        airflow_config["env"] = obj_in.env
+        return airflow_config
+
+    def _local_train_airflow_config_to_local_train_config_base(self, airflow_config, name):
+        config = airflow_config
+        config["name"] = name
+        config["image"] = self._get_image(config["repository"])
+        return config
+
+    def _get_repository(self, image: str):
+        """
+
+        @param image:
+        @return:
+        """
+
+        harbor_api = os.getenv("HARBOR_URL")
+        harbor_url = harbor_api.split("/")[2]
+        return f"{harbor_url}/{image}"
+
+    def _get_image(self, repository: str):
+        harbor_api = os.getenv("HARBOR_URL").split("/")[2]
+        image = repository.split(f"{harbor_api}/")[1]
+        return image
+
```

Comment:
i feel like this can and should ne signficantly simplified, i.e. change the input schema to include repository and tag.